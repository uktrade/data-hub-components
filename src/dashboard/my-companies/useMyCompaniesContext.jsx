import { orderBy, pick } from 'lodash'
import { useReducer } from 'react'
import createUseContext from 'constate'
import { FILTER_CHANGE, LIST_CHANGE, ORDER_CHANGE } from './constants'

export const filterCompanyName = (companies, filterText) =>
  filterText.length
    ? companies.filter((c) =>
        c.company.name.toLowerCase().includes(filterText.toLowerCase())
      )
    : companies

const defaultState = {
  lists: [],
  selectedIdx: 0,
  sortBy: 'recent',
  filter: '',
}

export const reducer = (state, { type, ...action }) => {
  switch (type) {
    case LIST_CHANGE:
      return { ...state, selectedIdx: action.idx }
    case FILTER_CHANGE:
      return { ...state, filter: action.filter }
    case ORDER_CHANGE:
      return { ...state, sortBy: action.sortBy }
    default:
      return state
  }
}

// We are unpacking children here just to remove them from initialState.
const useMyCompaniesContext = createUseContext(
  ({ children, editListsLinkProps, ...rest }) => {
    const initialState = pick(rest, [
      'lists',
      'selectedIdx',
      'sortBy',
      'filter',
    ])
    const [state, dispatch] = useReducer(reducer, {
      ...defaultState,
      ...initialState,
      // Reducer assumes the list of company lists is sorted.
      lists: orderBy(initialState.lists, 'name'),
    })

    return {
      state,
      dispatch,
      editListsLinkProps,
    }
  }
)

export default useMyCompaniesContext
