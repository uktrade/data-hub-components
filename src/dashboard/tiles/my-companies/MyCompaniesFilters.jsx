import React, { useState } from 'react'

function MyCompaniesFilters({ initialCompanies, companies, onFilterUpdate }) {
  const [filterCompanyName, setFilterCompanyName] = useState('')

  function filterByName(e) {
    const companyName = e.target.value
    setFilterCompanyName(companyName)

    const filteredCompanies = initialCompanies.filter(
      c => c.companyName.toUpperCase().includes(companyName.toUpperCase()),
    )
    onFilterUpdate(filteredCompanies)
  }

  return (
    <div>
      <input type="text" onChange={filterByName} value={filterCompanyName} />
    </div>
  )
}

export default MyCompaniesFilters
