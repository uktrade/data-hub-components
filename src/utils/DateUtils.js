import moment from 'moment'

const DATE_FORMAT_LONG = 'YYYY-MM-DD'
const DATE_FORMAT_SHORT = 'YYYY-MM'

function padZero(value) {
  const parsedValue = parseInt(value, 10)
  if (Number.isNaN(parsedValue)) {
    return value
  }
  return parsedValue < 10 ? `0${parsedValue}` : parsedValue.toString()
}

function normaliseAndFormatDate(year, month, day) {
  const y = padZero(year)
  const m = padZero(month)
  const yearAndMonth = `${padZero(y)}-${padZero(m)}`
  return day ? `${yearAndMonth}-${padZero(day)}` : yearAndMonth
}

export default class DateUtils {
  static format(dateStr) {
    return moment(dateStr).format('DD MMM YYYY')
  }

  static formatWithTime(dateTimeStr) {
    return moment(dateTimeStr).format('D MMM YYYY, h:mma')
  }

  static isDateValid = (year, month, day, format = DATE_FORMAT_LONG) => {
    const date = normaliseAndFormatDate(year, month, day)
    return moment(date, format, true).isValid()
  }

  static isShortDateValid(year, month) {
    return DateUtils.isDateValid(year, month, null, DATE_FORMAT_SHORT)
  }

  static transformValueForAPI({ year, month, day = 1 }) {
    if (year && month && day) {
      return normaliseAndFormatDate(year, month, day)
    }

    return null
  }
}
