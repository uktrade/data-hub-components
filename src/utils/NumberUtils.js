export default class NumberUtils {
  static decimal(number) {
    if (!number && number !== 0) {
      return null
    }

    const formatter = new Intl.NumberFormat('en-GB', {
      style: 'decimal',
    })

    return formatter.format(number)
  }

  static currencyGBP(number, options = { maximumSignificantDigits: 21 }) {
    if (!number && number !== 0) {
      return null
    }

    const formatter = new Intl.NumberFormat('en-GB', {
      style: 'currency',
      currency: 'GBP',
      ...options,
    })

    return formatter.format(number)
  }

  static currencyUSD(number, options = { maximumSignificantDigits: 21 }) {
    if (!number && number !== 0) {
      return null
    }

    const formatter = new Intl.NumberFormat('en-US', {
      style: 'currency',
      currency: 'USD',
      ...options,
    })

    return formatter.format(number)
  }

  static roundToSignificantDigits(number, sigDig) {
    if (!number && number !== 0) {
      return null
    }

    return Number(number.toPrecision(sigDig))
  }
}
