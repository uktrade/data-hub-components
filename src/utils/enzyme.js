export const withTargetValue = value => ({
  target: {
    value,
  },
})

/**
 * @function getOptionValueByText
 * @description Finds an `<option>` element with its inner text matching the
 * {text} parameter and returns the value of its `value` attribute.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {String} text - The option text
 */
export const getOptionValueByText = (wrapper, text) => (
  wrapper.find('option')
    .find({ children: text })
    .props()
    .value
)

/**
 * @function changeAndUpdate
 * @description Simulates the `change` event on the provided {wrapper} with
 * {vaule} as `target.value` and updates (re-renders) the wrapper.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {*} value - The value to be used as `event.target`
 */
export const changeAndUpdate = (wrapper, value) => {
  wrapper.simulate('change', withTargetValue(value))
  wrapper.update()
}

/**
 * @function changeSelectAndUpdate
 * @description Finds a `<select>` element in the wrapper, selects the option
 * matching the {text} and updates (re-renders) the wrapper.
 * @param {*} wrapper - Enzyme React wrapper
 * @param {String} text - The option text
 */
export const changeSelectAndUpdate = (wrapper, text) => {
  changeAndUpdate(wrapper.find('select'), getOptionValueByText(wrapper, text))
  wrapper.update()
}
