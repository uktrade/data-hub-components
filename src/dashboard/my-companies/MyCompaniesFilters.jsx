import React from 'react'
import styled from 'styled-components'
import { SelectInput } from '@govuk-react/select'
import LabelText from '@govuk-react/label-text'
import { MEDIA_QUERIES, SPACING } from '@govuk-react/constants'
import { typography } from '@govuk-react/lib'
import useMyCompaniesContext from './useMyCompaniesContext'
import { FILTER_CHANGE, ORDER_CHANGE } from './constants'

const StyledSelectInput = styled(SelectInput)(typography.font({ size: 14 }), {
  [MEDIA_QUERIES.LARGESCREEN]: {
    width: 'auto',
  },
})

const StyledLabelText = styled(LabelText)(typography.font({ size: 14 }))

const StyledContainer = styled.div`
  display: flex;
  justify-content: space-between;
`

const StyledSearch = styled.input`
  margin-right: ${SPACING.SCALE_2};
  padding-right: ${SPACING.SCALE_1};
  padding-left: ${SPACING.SCALE_1};
`

function MyCompaniesFilters() {
  const { dispatch } = useMyCompaniesContext()

  return (
    <StyledContainer>
      <StyledSearch
        type="text"
        placeholder="Search list"
        onChange={e => dispatch({
          type: FILTER_CHANGE,
          filter: e.target.value,
        })}
      />
      <StyledLabelText>
        Sort By:{' '}
        <StyledSelectInput
          onChange={e => dispatch({
            type: ORDER_CHANGE,
            sortBy: e.target.value,
          })}
        >
          <option value="recent">Recent interaction</option>
          <option value="least-recent">Least recent interaction</option>
          <option value="alphabetical">Company name A-Z</option>
        </StyledSelectInput>
      </StyledLabelText>
    </StyledContainer>
  )
}

export default MyCompaniesFilters
