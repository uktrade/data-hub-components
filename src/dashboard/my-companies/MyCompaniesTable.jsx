import { orderBy } from 'lodash'
import moment from 'moment'
import React from 'react'
import styled from 'styled-components'
import LinesEllipsis from 'react-lines-ellipsis'
import { typography } from '@govuk-react/lib'
import Table from '@govuk-react/table'
import Link from '@govuk-react/link'
import { GREY_1 } from 'govuk-colours'
import { MEDIA_QUERIES } from '@govuk-react/constants'
import useMyCompaniesContext from './useMyCompaniesContext'
import Filters from './MyCompaniesFilters'

const StyledCellHeader = styled(Table.CellHeader)(
  typography.font({ size: 14, weight: 'bold' }),
  {
    [MEDIA_QUERIES.DESKTOP]: {
      overflow: 'hidden',
      'white-space': 'nowrap',
    },
  }
)

const StyledDateCell = styled(Table.Cell)(typography.font({ size: 14 }), {
  color: GREY_1,
  'text-align': 'center',
})

export function sortCompanies(companies, sortType) {
  const sort = {
    recent: orderBy(companies, [c => c.latestInteraction.date || ''], ['desc']),
    'least-recent': orderBy(companies, [c => c.latestInteraction.date || ''], ['asc']),
    alphabetical: orderBy(companies, [c => c.company.name], ['asc']),
  }
  return sort[sortType]
}

export const filterCompanyName = (companies, filterText) => (filterText.length
  ? companies.filter(c => c.company.name.toLowerCase().includes(filterText.toLowerCase()))
  : companies)

function MyCompaniesTable() {
  const { state: { lists, selectedIdx, sortBy, filter } } = useMyCompaniesContext()
  const list = lists[selectedIdx]
  const filteredSortedCompanies = sortCompanies(
    filterCompanyName(list.companies, filter),
    sortBy,
  )

  const header = (
    <Table.Row>
      <StyledCellHeader>Company name</StyledCellHeader>
      <StyledCellHeader>Last interaction</StyledCellHeader>
      <StyledCellHeader>&nbsp;</StyledCellHeader>
    </Table.Row>
  )

  const rows = filteredSortedCompanies.map(({ company, latestInteraction }) => {
    return (
      <Table.Row key={company.id}>
        <Table.Cell>
          <Link href={`companies/${company.id}`}>
            <LinesEllipsis
              text={company.name}
              maxLine="2"
              ellipsis="..."
              trimRight={true}
              basedOn="words"
            />
          </Link>
        </Table.Cell>
        <StyledDateCell>
          {latestInteraction.date
            ? moment(latestInteraction.date).format('D MMM YYYY')
            : '-'
          }
        </StyledDateCell>
        <Table.Cell>
          {latestInteraction.id ? (
            <Link href={`interactions/${latestInteraction.id}`}>
              <LinesEllipsis
                text={latestInteraction.subject}
                maxLine="2"
                ellipsis="..."
                trimRight={true}
                basedOn="words"
              />
            </Link>
          ) : (
            latestInteraction.subject
          )}
        </Table.Cell>
      </Table.Row>
    )
  })

  return (
    <>
      {list.companies.length > 1 && <Filters />}
      <Table head={header}>{rows}</Table>
    </>
  )
}

export default MyCompaniesTable
