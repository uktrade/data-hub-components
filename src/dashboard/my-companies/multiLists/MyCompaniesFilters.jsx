import React, { useRef } from 'react'
import styled from 'styled-components'
import { SelectInput } from '@govuk-react/select'
import LabelText from '@govuk-react/label-text'
import { MEDIA_QUERIES, SPACING } from '@govuk-react/constants'
import { typography } from '@govuk-react/lib'
import ACTIONS from '../constants'
import useMyCompaniesContext from '../useMycompaniesListsContext'

const StyledSelectInput = styled(SelectInput)(typography.font({ size: 14 }), {
  [MEDIA_QUERIES.LARGESCREEN]: {
    width: 'auto',
  },
})

const StyledLabelText = styled(LabelText)(typography.font({ size: 14 }))

const StyledContainer = styled.div`
  display: flex;
  justify-content: space-between;
`

const StyledSearch = styled.input`
  margin-right: ${SPACING.SCALE_2};
  padding-right: ${SPACING.SCALE_1};
  padding-left: ${SPACING.SCALE_1};
`

function MyCompaniesFilters() {
  const { state, dispatch } = useMyCompaniesContext()
  const inputEl = useRef()

  const handleSelectChange = (e) => {
    dispatch({ type: ACTIONS.SORT_BY, sortType: e.target.value })
  }

  const handleInputChange = (e) => {
    dispatch({ type: ACTIONS.FILTER_BY, filterText: e.target.value })
  }

  return (
    <StyledContainer>
      <StyledSearch
        type="text"
        placeholder="Search list"
        onChange={handleInputChange}
        value={state.filterText}
        ref={inputEl}
      />
      <StyledLabelText>
        Sort By:{' '}
        <StyledSelectInput onChange={handleSelectChange} value={state.sortType}>
          <option value="recent">Recent interaction</option>
          <option value="least-recent">Least recent interaction</option>
          <option value="alphabetical">Company name A-Z</option>
        </StyledSelectInput>
      </StyledLabelText>
    </StyledContainer>
  )
}

export default MyCompaniesFilters
