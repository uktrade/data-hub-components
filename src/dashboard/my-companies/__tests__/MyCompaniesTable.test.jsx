import React from 'react'
import { mount } from 'enzyme'
import MyCompaniesTable from '../MyCompaniesTable'
import MyCompaniesFilters from '../MyCompaniesFilters'
import useMyCompaniesContext from '../useMyCompaniesContext'
import { changeAndUpdate, changeSelectAndUpdate } from '../../../utils/enzyme'

const getTableRowTexts = wrapper => (
  wrapper.find('tbody tr').map(row => row.text())
)

const fixture = companies => (
  mount(
    <useMyCompaniesContext.Provider lists={[
      {
        name: 'Lonely list',
        companies,
      },
    ]}
    >
      <MyCompaniesTable />
    </useMyCompaniesContext.Provider>,
  )
)

const companies = [
  {
    company: {
      name: 'Company A foo',
      id: 'a',
    },
    latestInteraction: {
      id: 'a',
      date: '2019-01-06',
      subject: 'Interaction A',
    },
  },
  {
    company: {
      name: 'Company B fooo',
      id: 'b',
    },
    latestInteraction: {
      id: 'a',
      date: '2019-01-04',
      subject: 'Interaction B',
    },
  },
  {
    company: {
      name: 'Company C foooo',
      id: 'c',
    },
    latestInteraction: {
      id: 'c',
      date: '2019-01-05',
      subject: 'Interaction C',
    },
  },
]

const fixtureMulti = () => fixture(companies)

describe('MyCompaniesTable', () => {
  describe('Single company', () => {
    const wrapper = fixture(companies.slice(0, 1))

    test('Should display a single row', () => {
      expect(getTableRowTexts(wrapper)).toEqual([
        'Company A foo6 Jan 2019Interaction A',
      ])
    })

    test('Should not display filters', () => {
      expect(wrapper.find(MyCompaniesFilters)).toHaveLength(0)
      expect(wrapper.text()).toBe(
        'Company nameLast interactionÂ Company A foo6 Jan 2019Interaction A',
      )
    })
  })

  describe('Multiple companies', () => {
    describe('Order by', () => {
      const wrapper = fixtureMulti()
      test('Should display filters', () => {
        expect(wrapper.find(MyCompaniesFilters)).toHaveLength(1)
      })

      test('Should order by recent interaction by default', () => {
        expect(getTableRowTexts(wrapper)).toEqual([
          'Company A foo6 Jan 2019Interaction A',
          'Company C foooo5 Jan 2019Interaction C',
          'Company B fooo4 Jan 2019Interaction B',
        ])
      })

      test('Should reorder on selecting the least recent order', () => {
        changeSelectAndUpdate(wrapper, 'Least recent interaction')
        expect(getTableRowTexts(wrapper))
          .toEqual([
            'Company B fooo4 Jan 2019Interaction B',
            'Company C foooo5 Jan 2019Interaction C',
            'Company A foo6 Jan 2019Interaction A',
          ])
      })

      test('Should reorder on selecting alphabetical sort option', () => {
        changeSelectAndUpdate(wrapper, 'Company name A-Z')
        expect(getTableRowTexts(wrapper))
          .toEqual([
            'Company A foo6 Jan 2019Interaction A',
            'Company B fooo4 Jan 2019Interaction B',
            'Company C foooo5 Jan 2019Interaction C',
          ])
      })
    })

    describe('Filter', () => {
      const wrapper = fixtureMulti()

      test('Should not apply any filters by default', () => {
        expect(getTableRowTexts(wrapper)).toEqual([
          'Company A foo6 Jan 2019Interaction A',
          'Company C foooo5 Jan 2019Interaction C',
          'Company B fooo4 Jan 2019Interaction B',
        ])
      })

      test('Should apply filters on filter input change', () => {
        changeAndUpdate(wrapper.find('input'), 'foo')
        expect(getTableRowTexts(wrapper)).toEqual([
          'Company A foo6 Jan 2019Interaction A',
          'Company C foooo5 Jan 2019Interaction C',
          'Company B fooo4 Jan 2019Interaction B',
        ])
      })
    })
  })
})
