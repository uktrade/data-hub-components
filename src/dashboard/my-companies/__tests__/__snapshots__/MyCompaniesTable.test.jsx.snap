// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCompaniesTable should match the snapshot 1`] = `
.c2 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c2:last-child {
  padding-right: 0;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 12px;
  line-height: 1.25;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

@media print {
  .c1 {
    font-size: 12px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 14px;
    line-height: 1.4285714285714286;
  }
}

@media only screen and (min-width:769px) {
  .c1 {
    overflow: hidden;
    white-space: nowrap;
  }
}

<Provider
  data={
    Array [
      Object {
        "company": Object {
          "id": "a1138c1c-d449-4846-aa58-18fae7e1cb92",
          "isArchived": false,
          "name": "A1 BMW LTD!!!!!",
        },
        "latestInteraction": Object {
          "date": "2019-08-14",
          "displayDate": "14 Aug 19",
          "id": "79d92719-7402-45b6-b3d7-eff559d6b282",
          "subject": "Here is a long interaction title some more text some more text some more text almost finished some more text nearly there more text finished",
        },
      },
      Object {
        "company": Object {
          "id": "a30b1db3-7140-44ca-ad7a-9824c3c2ed56",
          "isArchived": false,
          "name": "Portable Potatoes",
        },
        "latestInteraction": Object {
          "date": null,
          "displayDate": "-",
          "subject": "No interactions have been recorded",
        },
      },
      Object {
        "company": Object {
          "id": "b89b1db3-7140-44ca-ad7a-9824c3c2gh74",
          "isArchived": false,
          "name": "Zebra clothing",
        },
        "latestInteraction": Object {
          "date": "2019-02-21",
          "displayDate": "21 Feb 2019",
          "id": "86f92719-7402-45b6-b3d7-eff559d6b678",
          "subject": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.",
        },
      },
    ]
  }
>
  <MyCompaniesTable>
    <Table
      head={
        <ForwardRef>
          <ForwardRef>
            Company name
          </ForwardRef>
          <ForwardRef>
            Last interaction
          </ForwardRef>
          <ForwardRef>
            Â 
          </ForwardRef>
        </ForwardRef>
      }
    >
      <styled.table>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-EHOje",
                "isStatic": false,
                "lastClassName": "c0",
                "rules": Array [
                  "font-family: \\"nta\\", Arial, sans-serif; -webkit-font-smoothing: antialiased; -moz-osx-font-smoothing: grayscale; @media print {
  font-size: 14px; line-height: 1.15;
} font-weight: 400; font-size: 16px; line-height: 1.25; @media only screen and (min-width: 641px) {
  font-size: 19px; line-height: 1.3157894736842106;
}",
                  "color: #0b0c0c; @media print {
  color: #000;
}",
                  "width: 100%; border-spacing: 0; border-collapse: collapse;",
                  [Function],
                ],
              },
              "displayName": "styled.table",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-EHOje",
              "target": "table",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <table
            className="c0"
          >
            <styled.thead>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "bDWFJH",
                      "rules": Array [
                        "",
                      ],
                    },
                    "displayName": "styled.thead",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "thead",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <thead
                  className=""
                >
                  <styled.tr>
                    <StyledComponent
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-ifAKCX",
                            "isStatic": true,
                            "lastClassName": "cFlEyZ",
                            "rules": Array [
                              "",
                            ],
                          },
                          "displayName": "styled.tr",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-ifAKCX",
                          "target": "tr",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                    >
                      <tr
                        className=""
                      >
                        <Styled(CellHeader)>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-gzVnrw",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "font-family: \\"nta\\", Arial, sans-serif; -webkit-font-smoothing: antialiased; -moz-osx-font-smoothing: grayscale; @media print {
  font-size: 12px; line-height: 1.2;
} font-weight: 700; font-size: 12px; line-height: 1.25; @media only screen and (min-width: 641px) {
  font-size: 14px; line-height: 1.4285714285714286;
}",
                                    "@media only screen and (min-width: 769px) {
  overflow: hidden; white-space: nowrap;
}",
                                  ],
                                },
                                "displayName": "Styled(CellHeader)",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-gzVnrw",
                                "target": [Function],
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <CellHeader
                              className="c1"
                            >
                              <styled.td
                                as="th"
                                bold={true}
                                className="c1"
                                isHeader={true}
                              >
                                <StyledComponent
                                  as="th"
                                  bold={true}
                                  className="c1"
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-bxivhb",
                                        "isStatic": false,
                                        "lastClassName": "c2",
                                        "rules": Array [
                                          [Function],
                                          [Function],
                                          [Function],
                                        ],
                                      },
                                      "displayName": "styled.td",
                                      "foldedComponentIds": Array [],
                                      "propTypes": Object {
                                        "alignRight": [Function],
                                        "children": [Function],
                                        "isHeader": [Function],
                                        "numeric": [Function],
                                      },
                                      "render": [Function],
                                      "styledComponentId": "sc-bxivhb",
                                      "target": "td",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  isHeader={true}
                                >
                                  <th
                                    className="c1 c2"
                                  >
                                    Company name
                                  </th>
                                </StyledComponent>
                              </styled.td>
                            </CellHeader>
                          </StyledComponent>
                        </Styled(CellHeader)>
                        <Styled(CellHeader)>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-gzVnrw",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "font-family: \\"nta\\", Arial, sans-serif; -webkit-font-smoothing: antialiased; -moz-osx-font-smoothing: grayscale; @media print {
  font-size: 12px; line-height: 1.2;
} font-weight: 700; font-size: 12px; line-height: 1.25; @media only screen and (min-width: 641px) {
  font-size: 14px; line-height: 1.4285714285714286;
}",
                                    "@media only screen and (min-width: 769px) {
  overflow: hidden; white-space: nowrap;
}",
                                  ],
                                },
                                "displayName": "Styled(CellHeader)",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-gzVnrw",
                                "target": [Function],
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <CellHeader
                              className="c1"
                            >
                              <styled.td
                                as="th"
                                bold={true}
                                className="c1"
                                isHeader={true}
                              >
                                <StyledComponent
                                  as="th"
                                  bold={true}
                                  className="c1"
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-bxivhb",
                                        "isStatic": false,
                                        "lastClassName": "c2",
                                        "rules": Array [
                                          [Function],
                                          [Function],
                                          [Function],
                                        ],
                                      },
                                      "displayName": "styled.td",
                                      "foldedComponentIds": Array [],
                                      "propTypes": Object {
                                        "alignRight": [Function],
                                        "children": [Function],
                                        "isHeader": [Function],
                                        "numeric": [Function],
                                      },
                                      "render": [Function],
                                      "styledComponentId": "sc-bxivhb",
                                      "target": "td",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  isHeader={true}
                                >
                                  <th
                                    className="c1 c2"
                                  >
                                    Last interaction
                                  </th>
                                </StyledComponent>
                              </styled.td>
                            </CellHeader>
                          </StyledComponent>
                        </Styled(CellHeader)>
                        <Styled(CellHeader)>
                          <StyledComponent
                            forwardedComponent={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "attrs": Array [],
                                "componentStyle": ComponentStyle {
                                  "componentId": "sc-gzVnrw",
                                  "isStatic": true,
                                  "lastClassName": "c1",
                                  "rules": Array [
                                    "font-family: \\"nta\\", Arial, sans-serif; -webkit-font-smoothing: antialiased; -moz-osx-font-smoothing: grayscale; @media print {
  font-size: 12px; line-height: 1.2;
} font-weight: 700; font-size: 12px; line-height: 1.25; @media only screen and (min-width: 641px) {
  font-size: 14px; line-height: 1.4285714285714286;
}",
                                    "@media only screen and (min-width: 769px) {
  overflow: hidden; white-space: nowrap;
}",
                                  ],
                                },
                                "displayName": "Styled(CellHeader)",
                                "foldedComponentIds": Array [],
                                "render": [Function],
                                "styledComponentId": "sc-gzVnrw",
                                "target": [Function],
                                "toString": [Function],
                                "warnTooManyClasses": [Function],
                                "withComponent": [Function],
                              }
                            }
                            forwardedRef={null}
                          >
                            <CellHeader
                              className="c1"
                            >
                              <styled.td
                                as="th"
                                bold={true}
                                className="c1"
                                isHeader={true}
                              >
                                <StyledComponent
                                  as="th"
                                  bold={true}
                                  className="c1"
                                  forwardedComponent={
                                    Object {
                                      "$$typeof": Symbol(react.forward_ref),
                                      "attrs": Array [],
                                      "componentStyle": ComponentStyle {
                                        "componentId": "sc-bxivhb",
                                        "isStatic": false,
                                        "lastClassName": "c2",
                                        "rules": Array [
                                          [Function],
                                          [Function],
                                          [Function],
                                        ],
                                      },
                                      "displayName": "styled.td",
                                      "foldedComponentIds": Array [],
                                      "propTypes": Object {
                                        "alignRight": [Function],
                                        "children": [Function],
                                        "isHeader": [Function],
                                        "numeric": [Function],
                                      },
                                      "render": [Function],
                                      "styledComponentId": "sc-bxivhb",
                                      "target": "td",
                                      "toString": [Function],
                                      "warnTooManyClasses": [Function],
                                      "withComponent": [Function],
                                    }
                                  }
                                  forwardedRef={null}
                                  isHeader={true}
                                >
                                  <th
                                    className="c1 c2"
                                  >
                                    Â 
                                  </th>
                                </StyledComponent>
                              </styled.td>
                            </CellHeader>
                          </StyledComponent>
                        </Styled(CellHeader)>
                      </tr>
                    </StyledComponent>
                  </styled.tr>
                </thead>
              </StyledComponent>
            </styled.thead>
            <styled.tbody>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": true,
                      "lastClassName": "uTxCW",
                      "rules": Array [
                        "",
                      ],
                    },
                    "displayName": "styled.tbody",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "tbody",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <tbody
                  className=""
                />
              </StyledComponent>
            </styled.tbody>
          </table>
        </StyledComponent>
      </styled.table>
    </Table>
  </MyCompaniesTable>
</Provider>
`;
